Qualisys/QTM-Unity-Realtime-Streaming/Streaming/RTSkeleton.cs

```diff
@@ -15,21 +15,104 @@ public class RTSkeleton : MonoBehaviour
         private Dictionary<string, string> mMecanimToQtmSegmentNames = new Dictionary<string, string>();
 
         private HumanPoseHandler mSourcePoseHandler;
         private HumanPoseHandler mDestiationPoseHandler;
 
         protected RTClient rtClient;
         private Skeleton mQtmSkeletonCache;
+        private Skeleton mQtmSkeletonCacheCustom;
 
-        void Update()
+        private void UpdateCustom()
+        {
+
+            if (rtClient == null) rtClient = RTClient.GetInstance();
+
+            var skeleton = rtClient.GetSkeleton(SkeletonName);
+
+            if (mQtmSkeletonCacheCustom != skeleton)
+            {
+                mQtmSkeletonCacheCustom = skeleton;
+
+                if (mQtmSkeletonCacheCustom == null)
+                    return;
+
+                // User defined avatar/skeleton
+                mStreamedRootObjectCustom = null;
+
+                mQTmSegmentIdToGameObjectCustom = new Dictionary<uint, GameObject>(mQtmSkeletonCacheCustom.Segments.Count);
+
+                foreach (var segment in mQtmSkeletonCacheCustom.Segments.ToList())
+                {
+                    var gameObject = GameObject.Find(this.SkeletonName + ":" + segment.Value.Name);
+                    if (!gameObject)
+                        print("Didn't Find " + this.SkeletonName + ":" + segment.Value.Name);
+                    else
+                    {
+                        // First one is assumed to be the root
+                        if (mStreamedRootObjectCustom == null)
+                            mStreamedRootObjectCustom = gameObject;
+
+                        mQTmSegmentIdToGameObjectCustom[segment.Value.Id] = gameObject;
+                    }
+                }
+
+                if (mStreamedRootObjectCustom)
+                {
+                    mStreamedRootObjectCustom.transform.SetParent(this.transform, false);
+                }
+
+                return;
+            }
+
+            if (mQtmSkeletonCacheCustom == null)
+                return;
+
+            // Update all the game objects
+            foreach (var segment in mQtmSkeletonCacheCustom.Segments.ToList())
+            {
+                GameObject gameObject;
+                if (mQTmSegmentIdToGameObjectCustom.TryGetValue(segment.Key, out gameObject))
+                {
+                    Vector3 Position;
+                    Quaternion Rotation;
+
+                    if (segment.Value.ParentId == 0)
+                    {
+                        //Position = new Vector3(-segment.Value.Position.x / 1000, segment.Value.Position.z / 1000, -segment.Value.Position.y / 1000);
+                        Position = new Vector3(-segment.Value.Position.x / 1000, segment.Value.Position.y / 1000, segment.Value.Position.z / 1000);
+                        Position = rtClient.CoordinateSystemChange.Rotate(Position);
+
+                        Rotation = new Quaternion(0, 0, 0, 1);
+                        Rotation *= rtClient.CoordinateSystemChange;
```

Comment:
Isn't this the same as setting
Rotation = rtClient.CoordinateSystemChange
directly?
```suggestion
                        Rotation = rtClient.CoordinateSystemChange;
```